src/Jobs/Recruit.Vacancies.Jobs.UnitTests/Triggers/QueueTriggers/DeleteStaleVacanciesQueueTriggerTests.cs

```diff
@@ -0,0 +1,153 @@
+            _mockQuery.Verify(q => q.GetReferredVacanciesSubmittedBeforeAsync<VacancyIdentifier>(expectedReferredStaleByDate));
+        }
+        
+        [Fact]
+        public async Task WhenDateIsDefinedInTheMessage_ThenUseMessageDate()
+        {
+            var executionDate = new DateTime(2019, 10, 8);
+            _message.CreatedByScheduleDate = executionDate;
+
+            var expectedDraftStaleByDate = executionDate.AddDays(DeleteStaleVacanciesQueueTrigger.DefaultDraftStaleByDays * -1);
+            var expectedReferredStaleByDate = executionDate.AddDays(DeleteStaleVacanciesQueueTrigger.DefaultReferredStaleByDays * -1);
+
+            var sut = GetSut();
+            await sut.DeleteStaleVacanciesAsync(GetMessage(), null);
+
+            _mockTimeProvider.Verify(t => t.Today, Times.Never);
+            _mockQuery.Verify(q => q.GetDraftVacanciesCreatedBeforeAsync<VacancyIdentifier>(expectedDraftStaleByDate));
+            _mockQuery.Verify(q => q.GetReferredVacanciesSubmittedBeforeAsync<VacancyIdentifier>(expectedReferredStaleByDate));
+        }
+
+        [Fact]
+        public async Task WhenDraftAndReferredVacanciesFound_RaiseDeleteCommandForAll()
+        {
+            var fixture = new Fixture();
+            var draftVacancy = fixture.Create<VacancyIdentifier>();
+            var referredVacancy = fixture.Create<VacancyIdentifier>();
+            _mockQuery.Setup(q => q.GetDraftVacanciesCreatedBeforeAsync<VacancyIdentifier>(It.IsAny<DateTime>())).ReturnsAsync(new [] { draftVacancy });
+            _mockQuery.Setup(q => q.GetReferredVacanciesSubmittedBeforeAsync<VacancyIdentifier>(It.IsAny<DateTime>())).ReturnsAsync(new [] { referredVacancy });
+
+            var sut = GetSut();
+            await sut.DeleteStaleVacanciesAsync(GetMessage(), null);
+
+            _mockMessaging.Verify(m => m.SendCommandAsync(It.IsAny<ICommand>()), Times.Exactly(2));
+            _mockMessaging.Verify(m => m.SendCommandAsync(It.Is<DeleteVacancyCommand>(d => d.VacancyId == draftVacancy.Id)));
+            _mockMessaging.Verify(m => m.SendCommandAsync(It.Is<DeleteVacancyCommand>(d => d.VacancyId == referredVacancy.Id)));
+        }
+
+        [Fact]
+        public async Task WhenReferredVacanciesFound_RaiseDeleteCommand()
+        {
+            var fixture = new Fixture();
+            var vacancies = new [] { fixture.Create<VacancyIdentifier>() };
+            _mockQuery
+                .Setup(q => q.GetDraftVacanciesCreatedBeforeAsync<VacancyIdentifier>(It.IsAny<DateTime>()))
+                .ReturnsAsync(Array.Empty<VacancyIdentifier>());
+            _mockQuery
+                .Setup(q => q.GetReferredVacanciesSubmittedBeforeAsync<VacancyIdentifier>(It.IsAny<DateTime>()))
+                .ReturnsAsync(vacancies);
+
+            var sut = GetSut();
+            await sut.DeleteStaleVacanciesAsync(GetMessage(), null);
+
+            _mockMessaging.Verify(m => m.SendCommandAsync(It.IsAny<ICommand>()), Times.Exactly(1));
```

Comment:
The above is confirming the number of times generic SendCommandAsync has been invoked, notice the parameter type is ICommand. This will catch any future enhancements and force fixing the test.